src/tests/test_logged_activities.py

```diff
@@ -516,6 +516,178 @@ def test_secretary_edit_logged_activity_empty_payload(self):
+
+        """
+        Test a scenario where approval for logged activities fails
+        if the user is not a successops.
+        """
+
+        self.successops_role.save()
+        self.log_alibaba_challenge.status = 'pending'
+        self.log_alibaba_challenge2.status = 'rejected'
+        self.log_alibaba_challenge.save()
+        self.log_alibaba_challenge2.save()
+
+        self.payload = dict(
+            loggedActivitiesIds=[self.log_alibaba_challenge.uuid, self.log_alibaba_challenge2.uuid]
+        )
+
+        response = self.client.put(
+           f'/api/v1/approve/logged-activities',
+           data=json.dumps(self.payload),
+           headers=self.header
+        )
+
+        message = 'You\'re unauthorized to perform this operation'
+        response_details = json.loads(response.get_data(as_text=True))
+
+        self.assertEqual(response.status_code, 401)
+        self.assertEqual(response_details['message'], message)
+
+
+    def test_approving_logged_activities_empty_request_fails(self):
+
+        """
+        Test a scenario where approval for logged activities passes
+        if the user is a successops.
+        """
+
+        self.successops_role.save()
+
+        self.payload = dict(
+            loggedActivitiesIds=[self.log_alibaba_challenge.uuid, self.log_alibaba_challenge2.uuid]
+        )
+
+        response = self.client.put(
+           '/api/v1/approve/logged-activities',
+           headers=self.success_ops
+        )
+
+        message = 'Data for creation must be provided.'
+        response_details = json.loads(response.get_data(as_text=True))
+
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response_details['message'], message)
+
+
+    def test_approving_logged_activities_id_is_string_fails(self):
+
+        """
+        Test a scenario where approval for logged activities fails when
+        logged_activities_ids request payload is of type string.
+        """
+
+        self.successops_role.save()
+        self.payload = dict(
+            loggedActivitiesIds='400404004040'
+        )
+
+        response = self.client.put(
+           f'/api/v1/approve/logged-activities',
+           data=json.dumps(self.payload),
+           headers=self.success_ops
+        )
+
+        message = 'A List/Array with at least one logged activity id is needed!'
+        response_details = json.loads(response.get_data(as_text=True))
+
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response_details['message'], message)
+
+
+    def test_approving_over_twenty_logged_activities_fails(self):
+
+        """
+        Test a scenario where approval for logged activities fails
+        if logged activities request is more than 20.
+        """
+
+        self.successops_role.save()
+        self.payload = dict(
+            status='approved',
+            loggedActivitiesIds=['1','1','1','1','1','1','1','1','1','1','1','1','1','1','1','1',
```

Comment:
Try using a list comprehension for similar scenarios in the future:
`loggedActivtiesIds = ['i' for count in range(0,21)]`